src/main/webapp/app/overview/course-competencies/course-competencies-details.component.ts

```diff
@@ -107,15 +117,15 @@ export class CourseCompetenciesDetailsComponent implements OnInit, OnDestroy {
                     }
                 }
 
-                if (this.competency && this.competency.exercises) {
+                if (this.competency.exerciseLinks) {
                     // Add exercises as lecture units for display
-                    this.competency.lectureUnits = this.competency.lectureUnits ?? [];
-                    this.competency.lectureUnits.push(
-                        ...this.competency.exercises.map((exercise) => {
+                    this.competency.lectureUnitLinks = this.competency.lectureUnitLinks ?? [];
+                    this.competency.lectureUnitLinks.push(
+                        ...this.competency.exerciseLinks.map((exerciseLink) => {
                             const exerciseUnit = new ExerciseUnit();
-                            exerciseUnit.id = exercise.id;
-                            exerciseUnit.exercise = exercise;
-                            return exerciseUnit as LectureUnit;
+                            exerciseUnit.id = exerciseLink.exercise?.id;
+                            exerciseUnit.exercise = exerciseLink.exercise;
+                            return new CompetencyLectureUnitLink(this.competency, exerciseUnit as LectureUnit, 0.5);
```

Comment:
Could we please define this default value in a variable with a description why we choose 0.5?

Reply:
Done in #9564